apis/instanceha/v1beta1/instanceha_types.go

```diff
@@ -0,0 +1,150 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	condition "github.com/openstack-k8s-operators/lib-common/modules/common/condition"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/tls"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/util"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// Container image fall-back defaults
+
+	// InstanceHAContainerImage is the fall-back container image for InstanceHA
+	InstanceHAContainerImage = "quay.io/podified-antelope-centos9/openstack-openstackclient:current-podified"
+	OpenStackClould          = "default"
+)
+
+// InstanceHASpec defines the desired state of InstanceHA
+type InstanceHASpec struct {
+	// +kubebuilder:validation:Required
+	// +kubebuilder:default="quay.io/podified-antelope-centos9/openstack-openstackclient:current-podified"
+	// ContainerImage for the the InstanceHA container (will be set to environmental default if empty)
+	ContainerImage string `json:"containerImage"`
+
+        // +kubebuilder:validation:Required
+        // +kubebuilder:default="default"
+        // OpenStackClould is the name of the Cloud to use as per clouds.yaml (will be set to "default" if empty)
+        OpenStackCloud string `json:"openStackCloud"`
+
+	// +kubebuilder:validation:Required
+	// +kubebuilder:default=openstack-config
+	// OpenStackConfigMap is the name of the ConfigMap containing the clouds.yaml
+	OpenStackConfigMap *string `json:"openStackConfigMap"`
```

Comment:
I know we discussed and its a ptr in the openstackclient, but I don't think it makes sense to have it a ptr. I can not remember right now why I moved it to be a ptr in the openstackclient. I'd probably make it just a string if there is no reason to have it a ptr.